node/src/Pos/Client/CLI/Params.hs

```diff
@@ -112,14 +114,38 @@ getNodeParams cArgs@CommonNodeArgs{..} NodeArgs{..} systemStart = do
         , ..
         }
 
-getTransportParams :: CommonNodeArgs -> NetworkConfig kademlia -> TransportParams
-getTransportParams args networkConfig = TransportParams { tpTcpAddr = tcpAddr }
-  where
-    tcpAddr = case ncTopology networkConfig of
-        TopologyBehindNAT{} -> TCP.Unaddressable
-        _ -> let (bindHost, bindPort) = bindAddress args
-                 (externalHost, externalPort) = externalAddress args
-                 tcpHost = BS8.unpack bindHost
-                 tcpPort = show bindPort
-                 tcpMkExternal = const (BS8.unpack externalHost, show externalPort)
-             in  TCP.Addressable $ TCP.TCPAddrInfo tcpHost tcpPort tcpMkExternal
+data NetworkTransportMisconfiguration =
+
+      -- | A bind address was not given.
+      MissingBindAddress
+
+      -- | An external address was not given.
+    | MissingExternalAddress
+
+      -- | An address was given when one was not expected (behind NAT).
+    | UnnecessaryAddress
+
+instance Show NetworkTransportMisconfiguration where
+    show MissingBindAddress     = "No network bind address given"
+    show MissingExternalAddress = "No external network address given"
+    show UnnecessaryAddress     = "Network address given when none was expected"
```

Comment:
I realise it feels non-modular to mention flag names here, but error messages are a bit non-modular. In theory we'd install exception handlers near the program top and it'd use the full context to format error messages (where it knows what the full CLI is etc). But since we're not going to those lengths then I think an error message that knows a bit too much about another part of the impl isn't that bad.